File: java/ooad/TicTacToeApp/src/com/techlabs/tictactoe/test/TestResultAnalyzer.java
package com.techlabs.tictactoe.test;

import static org.junit.Assert.*;

import org.junit.Test;

import com.techlabs.tictactoe.Board;
import com.techlabs.tictactoe.MarkType;
import com.techlabs.tictactoe.Player;
import com.techlabs.tictactoe.ResultAnalyzer;

public class TestResultAnalyzer {

	@Test
	public void testWinResult() {
		Board board=new Board();
		ResultAnalyzer resultAnalyzer=new ResultAnalyzer(board);
		
		board.updateBoard(MarkType.NOUGHT, 4);
		board.updateBoard(MarkType.CROSS, 1);
		board.updateBoard(MarkType.NOUGHT, 1);
		board.updateBoard(MarkType.NOUGHT, 6);
		board.updateBoard(MarkType.CROSS, 7);
		board.updateBoard(MarkType.NOUGHT, 5);
		board.displayBoard();
		
		resultAnalyzer.analyzeBoard();
		System.out.println(resultAnalyzer.getResult().toString());;
	}
	
	@Test
	public void testDrawResult() {
		Board board=new Board();
		ResultAnalyzer resultAnalyzer=new ResultAnalyzer(board);
		
		board.updateBoard(MarkType.CROSS, 1);
		board.updateBoard(MarkType.NOUGHT, 2);
		board.updateBoard(MarkType.CROSS, 3);
		board.updateBoard(MarkType.NOUGHT, 4);
		board.updateBoard(MarkType.CROSS, 5);
		board.updateBoard(MarkType.NOUGHT, 9);
		board.updateBoard(MarkType.CROSS, 8);
		board.updateBoard(MarkType.NOUGHT, 7);
		board.updateBoard(MarkType.CROSS, 6);
		board.displayBoard();
		
		resultAnalyzer.analyzeBoard();
		System.out.println(resultAnalyzer.getResult().toString());;
	}
	
	@Test
	public void testOnGoingResult() {
		Board board=new Board();
		ResultAnalyzer resultAnalyzer=new ResultAnalyzer(board);
		
		board.updateBoard(MarkType.CROSS, 1);
		board.updateBoard(MarkType.NOUGHT, 2);
		board.updateBoard(MarkType.CROSS, 3);
		board.updateBoard(MarkType.NOUGHT, 4);
		board.updateBoard(MarkType.CROSS, 5);
		board.updateBoard(MarkType.NOUGHT, 6);
		board.displayBoard();
		
		resultAnalyzer.analyzeBoard();
		System.out.println(resultAnalyzer.getResult().toString());;
	}

}

File: dotNet/ClassLibrary/WinForm/TicTacToeApp/TicTaeToeUnitTest/TestResultAnalyzer.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using TicTacToeCore;

namespace TicTaeToeUnitTest
{
    [TestClass]
    public class TestResultAnalyzer
    {
        [TestMethod]
        public void TestWinResult()
        {

        }

        [TestMethod]
        public void TestDrawResult()
        {

        }

        [TestMethod]
        public void TestOnGoingResult()
        {
            Board board = new Board();
            ResultAnalyzer resultAnalyzer = new ResultAnalyzer(board);

            MarkType cross = MarkFactory.CROSS;
            MarkType nought = MarkFactory.NOUGHT;

            board.UpdateBoard(nought, 1);
            board.UpdateBoard(cross, 8);
            board.UpdateBoard(cross, 9);

            resultAnalyzer.AnalyzeBoard();

            Assert.AreEqual(Result.ONGOING, resultAnalyzer.Result);
        }

        [TestMethod]
        public void TestVerticalCheckWinMethod()
        {
            Board board = new Board();
            ResultAnalyzer resultAnalyzer = new ResultAnalyzer(board);

            MarkType cross = MarkFactory.CROSS;
            MarkType nought = MarkFactory.NOUGHT;

            board.UpdateBoard(cross, 3);
            board.UpdateBoard(cross, 6);
            board.UpdateBoard(cross, 9);

            resultAnalyzer.AnalyzeBoard();

            Assert.AreEqual(Result.WIN, resultAnalyzer.Result);
        }

        [TestMethod]
        public void TestHorizontalCheckWinMethod()
        {
            Board board = new Board();
            ResultAnalyzer resultAnalyzer = new ResultAnalyzer(board);

            MarkType cross = MarkFactory.CROSS;
            MarkType nought = MarkFactory.NOUGHT;

            board.UpdateBoard(cross, 7);
            board.UpdateBoard(cross, 8);
            board.UpdateBoard(cross, 9);

            resultAnalyzer.AnalyzeBoard();

            Assert.AreEqual(Result.WIN, resultAnalyzer.Result);
        }

        [TestMethod]
        public void TestDiagonalCheckWinMethod()
        {
            Board board = new Board();
            ResultAnalyzer resultAnalyzer = new ResultAnalyzer(board);

            MarkType cross = MarkFactory.CROSS;
            MarkType nought = MarkFactory.NOUGHT;

            board.UpdateBoard(cross, 3);
            board.UpdateBoard(cross, 5);
            board.UpdateBoard(cross, 7);

            resultAnalyzer.AnalyzeBoard();

            Assert.AreEqual(Result.WIN, resultAnalyzer.Result);
        }
    }
}
